pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -95,6 +96,11 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		tr, err := r.TaskRunLister.TaskRuns(pr.Namespace).Get(name)
 		if err != nil {
 			logging.FromContext(ctx).Errorf("Unable to get reconciled status of taskrun %s within pipelinerun", name)
+			if errors.IsNotFound(err) {
```

Comment:
Can we add a test to make sure that we don't try and write an attestation for a PipelineRun that has a missing TaskRun?

We probably should just drop the PipelineRun and refuse to sign it.

Reply:
Thanks. Added the test. Let me know what you think.

Reply:
Looks great! Thanks.